### PR TITLE
Resolve the country code from the geographic code

### LINKED ISSUE #58: Country codes are incorrect & incorrect length
**Type of change**
Change to existing mapping

**Example RDF records**
https://api.stage.sinopia.io/resource/7696c434-8b95-474d-976e-e9b1e78306ef

image
(I think this is where it is derived from...)

**Resource template**
https://stage.sinopia.io/editor/pcc:bf2:Monograph:Instance with referred resource template https://api.stage.sinopia.io/resource/pcc:bf2:Provision:Name

**Expected MARC**
008/15-17 (Place of production, production, or execution) gau
This is from the MARC country codes at https://id.loc.gov/vocabulary/countries.html (coding, not label). A 3-letter code is required.

**Actual MARC**
us 
This is intended to be "U.S.", but is not correct (it would be "xxu"--US states have 2 letters for the state and one letter for the US. If state is unknown "xx" is used instead. This is also used with Canadian provinces and Australian states) since it derived from the 043 codes rather than the MARC country codes. The incorrect length makes the MARC record invalid and it cannot be loaded to OCLC.

**Expected mapping logic**
This is difficult, because at the moment we have no way to code the country codes directly. We could do that by replacing the place of publication lookup to the country codes. There may be a way to derive the country/state/province by the abbreviated term in parentheses after the city name.

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?
Fixes #58

### CODE REVIEW

File: lib/rdf2marc/model2marc/control_field_008.rb [hunk 1]
```diff
@@ -30,6 +30,29 @@ def value
...
+      # @param [String] code from https://www.loc.gov/marc/geoareas/gacs_code.html
+      # @return [String] a three letter code from https://id.loc.gov/vocabulary/countries.html
+      def place_from_geo_area_code(code)
+        return 'xx ' unless code
+        return lookup_country_from_geoarea(code) if in_country_authority?(code)
+
+        code[0, 3]
+      end
+
+      def lookup_country_from_geoarea(code)
```
Comment (justinlittman) [inline lib/rdf2marc/model2marc/control_field_008.rb:42]: Lookups / resolving should be done as part of rdf2model, not model2marc.

**Review by justinlittman — CHANGES_REQUESTED** [review]

File: lib/rdf2marc/model2marc/control_field_008.rb (hunk 1 above)
Comment (jcoyne) [reply lib/rdf2marc/model2marc/control_field_008.rb:42]: This seems like an important design decision. Is this documented somewhere?

Comment (justinlittman) [reply lib/rdf2marc/model2marc/control_field_008.rb:42]: In the code? (I'm open to other suggestions).

All resolving/lookup (such as this) is handled by a Resolver, in this case https://github.com/LD4P/rdf2marc/blob/main/lib/rdf2marc/resolver/id_loc_gov_resolver.rb.

That resolving is performed in rdf2model, for example, https://github.com/LD4P/rdf2marc/blob/main/lib/rdf2marc/rdf2model/mappers/number_and_code_fields.rb#L105.

Comment (jcoyne) [reply lib/rdf2marc/model2marc/control_field_008.rb:42]: I was thinking of a design doc.

**Review by justinlittman — APPROVED** [review]

**Pull request merged**